docs/src/main/docs/security/05_extensibility.adoc

```diff
@@ -0,0 +1,117 @@
+///////////////////////////////////////////////////////////////////////////////
+
+    Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+///////////////////////////////////////////////////////////////////////////////
+
+= Extending Security
+
+This guide guides you through extension possibilities for Security component.
+
+The component has the following extension points:
+
+* Security providers
+* Provider selection policy
+* Integration with a framework
+
+== Security providers
+You can build a custom provider for each type of security concept supported.
+By default, each provider is asynchronous. To simplify simple cases, a class
```

Comment:
"...simplify simple case..." sounds odd. How about "For simple use cases, ..." ?